docs/generation.rst

```diff
@@ -3,122 +3,91 @@
-
-.. code-block:: bash
-
-  $ npm install edgedb       # npm users
-  $ yarn add edgedb          # yarn users
-
-Then install ``@edgedb/generate`` as a dev dependency.
+Run a generator with the following command.
 
-.. code-block:: bash
+.. tabs::
 
-  $ npm install @edgedb/generate --save-dev      # npm users
-  $ yarn add @edgedb/generate --dev              # yarn users
+  .. code-tab:: bash
+    :caption: npm
 
-Running a generator
-^^^^^^^^^^^^^^^^^^^
+    $ npx @edgedb/generate <generator> [options]
 
-Run a generator with the following command.
+  .. code-tab:: bash
+    :caption: yarn
 
-**Node.js**
+    $ yarn run -B generate <generator> [options]
 
-.. code-block:: bash
+  .. code-tab:: bash
+    :caption: pnpm
 
-  $ npx @edgedb/generate <generator> [options]
+    $ pnpm exec generate <generator> [options]
 
+  .. code-tab:: bash
+    :caption: Deno
 
-**Deno**
+    $ deno run \
+      --allow-all \
+      --unstable \
+      https://deno.land/x/edgedb/generate.ts <generator> [options]
 
-.. code-block:: bash
+The value of generators should be one of the following:
 
-  $ deno run --allow-all --unstable https://deno.land/x/edgedb/generate.ts <generator> [options]
+.. list-table::
+   :class: funcoptable
 
+   * - ``edgeql-js``
+     - Provides a **code-first** way to write **fully-typed** EdgeQL queries
+       with TypeScript. We recommend it for TypeScript users, or anyone who
+       prefers writing queries with code.
+     - :ref:`docs <edgedb-js-qb>`
 
-The value of ``<generator>`` should be one of the following.
+   * - ``queries``
+     - Scans your project for ``*.edgeql`` files and generates functions that
+       allow you to execute these queries in a typesafe way.
+     - :ref:`docs <edgedb-js-queries>`
 
-- ``queries``: This generator scans your project for ``*.edgeql`` files and generates a file containing a strongly-typed function alongside each. Alternatively you can use ``--file`` mode to generate a single file containing all the "query functions".
-- ``edgeql-js``: This generator introspects your database schema and generates a query builder.
-- ``interfaces``: This generator introspects your database schema and generates TypeScript interfaces for each object type.
+   * - ``interfaces``
+     - Introspects your schema and generates file containing *TypeScript
+       interfaces* that correspond to each object type. This is useful for
+       writing typesafe code to interact with EdgeDB.
+     - :ref:`docs <edgedb-js-interfaces>`
 
 Connection
 ^^^^^^^^^^
 
-Generating the query builder requires a
-connection to an active EdgeDB database. It does **not** simply read your local ``.esdl`` schema files. Generators rely on the database to introspect the schema and analyze queries. Doing so without a database connection would require implementing a full EdgeQL parser and static analyzer in JavaScript—which we don't intend to do anytime soon.
+Generating the query builder requires a connection to an active EdgeDB database.
```

Comment:
Is this true only for the query builder or for all generators?

Reply:
True for all: we introspect the database in all three generators.